frontend/components/wallet/WalletProvider.tsx

```diff
@@ -40,6 +41,11 @@ export const WalletProvider: React.FC<Props> = ({ children }) => {
   );
   const toast = useToast();
 
+  useEffect(() => {
+    const initProvider = getProvider();
```

Comment:
since this function relies on connectedChainId and mainUPController shall we make them parameters and pass them here?